sql/core/src/main/java/org/apache/spark/sql/sources/v2/reader/streaming/ContinuousPartitionReaderFactory.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.sources.v2.reader.streaming;
+
+import java.io.Serializable;
+
+import org.apache.spark.annotation.InterfaceStability;
+import org.apache.spark.sql.catalyst.InternalRow;
+import org.apache.spark.sql.sources.v2.reader.InputPartition;
+import org.apache.spark.sql.vectorized.ColumnarBatch;
+
+/**
+ * A factory used to create {@link ContinuousPartitionReader} instances.
+ */
+@InterfaceStability.Evolving
+public interface ContinuousPartitionReaderFactory extends Serializable {
```

Comment:
@rdblue I have to fork a `PartitionReaderFactory` for continuous streaming, in order to use the name `createContinuousReader`. Also I need to rename `ContinuousReaderSupport.createReaderFactory ` to `createContinuousReaderFactory`, which requires to move out `createReaderFactory` from the base interace `ReadSupport`.

Is it really worth? The existing sources only implement `XYZReadSupportProvider` at the same time, but not `XYZReadSupport`. e.g. micro-batch and continuous read support are always 2 different classes, as the logic is so different. I believe file source will be the same.

Reply:
That said, the data source implementations probably do not care if they can implement different read support at the same time, but we need to pay a high cost for this change.

Reply:
It should be fine to use the same method name in different `ReaderFactory` interfaces. The problem I identified was when all of the `ReadSupport` interfaces used the same `createReaderFactory` method to create different factories. Implementing multiple `ReadSupport` interfaces on the same table would require that these be the same method and therefore return the same class, which would be forced to support all `ReaderFactory` uses.

If construction of the reader factory happens with different methods in each `ReadSupport`, then it is up to the implementation whether they are the same class or different ones. They can use the same method name to create readers in that case.